pkg/project/options.go

```diff
@@ -25,58 +25,81 @@ import (
 	"github.com/pterm/pterm"
 
 	"github.com/nitrictech/cli/pkg/runtime"
+	"github.com/nitrictech/cli/pkg/stack"
 	"github.com/nitrictech/cli/pkg/utils"
 )
 
 var stackPath string
 
-func FromConfig(p *Config) (*Project, error) {
-	s := New(p)
+func FromConfig(c *Config, s *stack.Config) (*Project, error) {
+	p := New(c)
 
-	for _, g := range p.Handlers {
-		maybeFile := filepath.Join(s.Dir, g)
+	for _, h := range c.Handlers {
+		maybeFile := filepath.Join(p.Dir, h)
 
 		if _, err := os.Stat(maybeFile); err != nil {
-			fs, err := utils.GlobInDir(stackPath, g)
+			fs, err := utils.GlobInDir(stackPath, h)
 			if err != nil {
 				return nil, err
 			}
 
 			for _, f := range fs {
-				fn, err := FunctionFromHandler(f, s.Dir)
+				fn, err := FunctionFromHandler(f, s)
 				if err != nil {
 					return nil, err
 				}
 
-				s.Functions[fn.Name] = fn
+				p.Functions[fn.Name] = fn
 			}
 		} else {
-			fn, err := FunctionFromHandler(g, s.Dir)
+			fn, err := FunctionFromHandler(h, s)
 			if err != nil {
 				return nil, err
 			}
 
-			s.Functions[fn.Name] = fn
+			p.Functions[fn.Name] = fn
 		}
 	}
 
-	if len(s.Functions) == 0 {
-		return nil, fmt.Errorf("no functions were found with the glob '%s', try a new pattern", strings.Join(p.Handlers, ","))
+	if len(p.Functions) == 0 {
+		return nil, fmt.Errorf("no functions were found with the glob '%s', try a new pattern", strings.Join(c.Handlers, ","))
 	}
 
-	return s, nil
+	return p, nil
 }
 
-func FunctionFromHandler(h, stackDir string) (Function, error) {
+func FunctionFromHandler(h string, s *stack.Config) (Function, error) {
 	pterm.Debug.Println("Using function from " + h)
 
 	rt, err := runtime.NewRunTimeFromHandler(h)
 	if err != nil {
 		return Function{}, err
 	}
 
+	c, ok := s.Config[h]
```

Comment:
this ^ is not smart and needs the full name of the handler.. could be improved, also i don't have the "default" case yet..